benchmarks/definition/core.py

```diff
@@ -0,0 +1,66 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from datetime import datetime
+from typing import Generic
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of, is_callable
+from pandas import DataFrame
+
+from benchmarks.definition.config import BenchmarkConfig
+from benchmarks.result.metadata_class import ResultMetadata
+from benchmarks.result.result import Result
+
+BenchmarkFunction = Callable[[BenchmarkConfig], DataFrame]
+
+
+@define
+class Benchmark(Generic[BenchmarkConfig]):
+    """A class to define a benchmark task."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
+    """The unique identifier of the benchmark running which can be set
+    to compare different executions of the same benchmark setting."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    callable: BenchmarkFunction = field(validator=is_callable())
+    """The function that executes the benchmark code and returns
+    the results as well as metadata."""
+
+    settings: BenchmarkConfig | None = field(default=None)
```

Comment:
@AVHopp @Scienfitz: Settings or Config, what do you prefer?

Reply:
indifferent